common/test/common/RequestLoggerTest.scala

```diff
@@ -42,11 +44,32 @@ class RequestLoggerTest extends AnyFlatSpec with Matchers {
     notExpectedFields.forall(fields.toList.contains) should be(false)
   }
 
+  "RequestLogger with request" should "tolerate case-insensitive HTTP headers names, outputting standard case regardless" in {
+    def loggedFieldsFor(headers: (String, String)*) =
+      RequestLoggerFields(
+        request = Some(baseRequest.withHeaders(headers: _*)),
+        response = None,
+        stopWatch = None,
+      ).toList
+
+    val logfieldWithStandardCase = LogFieldString("req.header.User-Agent", "Example-UA")
+    loggedFieldsFor("User-Agent" -> "Example-UA") should contain(logfieldWithStandardCase)
+    loggedFieldsFor("User-agent" -> "Example-UA") should contain(logfieldWithStandardCase)
+    loggedFieldsFor("user-agent" -> "Example-UA") should contain(logfieldWithStandardCase)
+    loggedFieldsFor("USER-AGENT" -> "Example-UA") should contain(logfieldWithStandardCase)
+  }
+
   "RequestLogger with response" should "log expected fields" in {
     val fields = RequestLoggerFields(request = None, response = Some(Ok), stopWatch = None)
     val expectedFields: List[LogField] = List(
       "resp.status" -> 200,
     )
     expectedFields.forall(fields.toList.contains) should be(true)
   }
+
+  "RequestLogger with response" should "tolerate case-insensitive HTTP headers names" in {
```

Comment:
We're introducing a similar unit test to ensure the response header names can be case-insensitive as well and it seems this is already happening.